### PR TITLE
Calling video comparison to improve the security strength

### PR DESCRIPTION
**Pull request by oscar-davids**
**Calling video comparison functions on results received from different Os**

**Specific updates (required)**

-  Add downloading function [here](https://github.com/livepeer/go-livepeer/blob/cddc3fffe93fe01e1a361bf7a7e0dea5a96dc00a/server/broadcast.go#L488) to download segmToCheckIndex-th trusted segment.
- Also [here ](https://github.com/livepeer/go-livepeer/blob/cddc3fffe93fe01e1a361bf7a7e0dea5a96dc00a/server/broadcast.go#L500)call GetSegmentData function to download untrusted segment.
- Call [CompareVideoByBuffer ](https://github.com/livepeer/lpms/blob/44fe3cdd279054f6ff399b223d01ed19f10c9e17/ffmpeg/ffmpeg.go#L233)to compare two segments before calling [CompareSignatureByBuffer](https://github.com/livepeer/go-livepeer/blob/cddc3fffe93fe01e1a361bf7a7e0dea5a96dc00a/server/broadcast.go#L506).

**How did you test each of these updates (required)**

- Call video comparison function in `chooseResults()`.
- Passed Unit Test.

**Does this pull request close any open issues?**

Fix [#315](https://github.com/livepeer/internal-project-tracking/issues/315)

**Checklist:**

- [X] Read the [contribution guide](./doc/contributing.md)
- [X] `make` runs successfully
- [X] All tests in `./test.sh` pass
- [ ] README and other documentation updated
- [X] [Pending changelog](./CHANGELOG_PENDING.md) updated

### CODE REVIEW

File: server/push_test.go [hunk 1]
```diff
@@ -1533,7 +1533,9 @@ func TestPush_ReuseIntmidWithDiffExtmid(t *testing.T) {
 
 func TestPush_MultipartReturnMultiSession(t *testing.T) {
 	assert := assert.New(t)
-
+	//need real video data for fast verification
```
Comment (leszko) [inline server/push_test.go]:
```suggestion
	// need real video data for fast verification
```

File: server/push_test.go [hunk 2]
```diff
@@ -1533,7 +1533,9 @@ func TestPush_ReuseIntmidWithDiffExtmid(t *testing.T) {
...
+	transcodeddata, err := ioutil.ReadFile("../core/test.ts")
```
Comment (leszko) [inline server/push_test.go:1537]: Does it test the fast verification? If yes, then which check `pHash` or `CompareVideoByBuffer`? I think we should have somewhere tests for both.

**Review by leszko — COMMENTED** [review]
Looks, good! Good work @oscar-davids 

Added 2 minor comments. Other than that, here are some comments things to consider:
1. Traffic-wise, we add downloading the whole segment. I guess it's ok, but just want to double-check it.
2. One optimization would be to check first `pHash` and only if `pHash` matches, then download the segment. But not sure we need it because we actually expect `pHash` to match in most cases.
3. One other optimization could be that we actually download the segment from untrusted O twice. One for the fast verification and later to download the result [here](https://github.com/livepeer/go-livepeer/blob/985d0ec65659c0586cf1abdce1ac5e639e6d9b8f/server/broadcast.go#L1112). Again, maybe we don't need to change it right now.
4. I didn't check if `ffmpeg.CompareVideoByBuffer()` works fine and the implication on having some false-negative results.

File: server/push_test.go (hunk 2 above)
Comment (oscar-davids) [reply server/push_test.go:1537]: Will add unit test  for `CompareVideoByBuffer`  separately.

**Comment by oscar-davids** [comment]
> 1. Traffic-wise, we add downloading the whole segment. I guess it's ok, but just want to double-check it.

Yeah, there is additional traffic for downloading the random n-th segment.

> 2. One optimization would be to check first `pHash` and only if `pHash` matches, then download the segment. But not sure we need it because we actually expect `pHash` to match in most cases.
> 3. One other optimization could be that we actually download the segment from untrusted O twice. One for the fast verification and later to download the result here. Again, maybe we don't need to change it right now.

I prefer the second optimization path because I think it is a little late to check validity at that time [here](https://github.com/livepeer/go-livepeer/blob/985d0ec65659c0586cf1abdce1ac5e639e6d9b8f/server/broadcast.go#L1112) with the third way.

> 4. I didn't check if `ffmpeg.CompareVideoByBuffer()` works fine and the implication on having some false-negative results.

For H264 encoding, it showed 100% accuracy in GPU vs CPU cross-comparison. I think we need to re-evaluate the H265 codec.

**Review by AlexKordic — APPROVED** [review]
Well Done!

**Review by leszko — APPROVED** [review]
LGTM 👍 

Thanks for adding the unit test!

**Pull request merged**